catalog/task/add-bundles-to-image-index/0.1/add-bundles-to-image-index.yaml

```diff
@@ -0,0 +1,110 @@
+    - name: IIB_BUILD_TIMEOUT_SECONDS
+      type: string
+      description:  Timeout seconds to receive the build state
+  results:
+    - name: json_build_info
+      description: JSON build information for the requested build
+    - name: build_state
+      description: IIB Service build state
+  steps:
+    - name: add-bundles-to-image-index
+      image: quay.io/hacbs-release/release-utils
+      script: |
+        #!/usr/bin/env bash
+        #
+        dnf -y install krb5-workstation > /dev/null
+
+        # adds the json request parameters to a file to be used as input data
+        # for curl and preventing shell expansion.
+        json_input=/tmp/$$.tmp
+        cat > $json_input <<JSON
+        $(params.IIB_JSON_BUILD_ARGS)
+        JSON
+
+        # replaces {TOKEN} placeholder with credentials.
+        if [ -f /mnt/service-secret/registry_token ]; then
+            CREDS=$(cat /mnt/service-secret/registry_token)
+            sed -i "s/{TOKEN}/${CREDS}/g" ${json_input}
+        fi
+
+        # performs kerberos authentication.
+        export KRB5_CONFIG=/mnt/service-secret/krb5.conf
+        /usr/bin/kinit $(cat /mnt/service-secret/krb5_username) < /mnt/service-secret/krb5_password
+
+        # adds image to the index.
+        /usr/bin/curl -u : --negotiate -s -X POST -H "Content-Type: application/json" -d@${json_input} --insecure \
+        "$(params.IIB_URL)/builds/add" | jq | tee $(results.json_build_info.path)
+
+        # checks if the previous call returned an error.
+        ! jq -e -r ".error | select( . != null )" $(results.json_build_info.path)
+      volumeMounts:
+        - name: service-secret
+          mountPath: /mnt/service-secret
+    - name: wait-for-build-state
+      image: quay.io/hacbs-release/release-utils
+      script: |
+        #!/usr/bin/env bash
+        #
+        TASKRUN="/tmp/$$.sh"
+        cat > ${TASKRUN} <<SH
+        #!/usr/bin/env bash
+        #
+        build_id=`jq -r ".id" $(results.json_build_info.path)`
+        state=""
+        while true; do
+            #
+            # fetching build information.
+            build_info=\$(/usr/bin/curl -s --insecure "$(params.IIB_URL)/builds/\${build_id}")
+
+            # get state from the build information.
+            state=\$(jq -r ".state" <<< \${build_info})
+            case \${state} in
+              "complete") break ;;
+              "failed") break ;;
+              *) echo -en "."; sleep 30; continue ;;
+            esac
+        done
+        echo
+        jq '{ "state": .state, "state_reason": .state_reason }' <<< \${build_info} | tee $(results.build_state.path)
+        test \${state} = "complete" && exit 0 || exit 1
+        SH
+
+        chmod +x ${TASKRUN}
+        echo -en "waiting for build state to exit..."
+
+        # adding timeout here due to the Task timeout not accepting $(params.IIB_BUILD_TIMEOUT_SECONDS)
+        # as parameter.
+        timeout $(params.IIB_BUILD_TIMEOUT_SECONDS) ${TASKRUN}
```

Comment:
https://tekton.dev/vault/pipelines-v0.19.0/tasks/#specifying-a-timeout

Reply:
I saw that. It can only work if I specify it literally. If I use the `timeout: "$(params.IIB_BUILD_TIMEOUT_SECONDS)s"` (or any combination) it breaks the task @davidmogar.

Reply:
That's unbelievable... so bad.